docs/server/source/data-models/inputs-outputs.rst

```diff
@@ -1,153 +1,102 @@
+At a high level, a condition is like a lock on an output.
+If can you satisfy the condition, you can unlock the output and transfer/spend it.
+BigchainDB Server v1.0 supports a subset of the ILP Crypto-Conditions
+(`version 02 of Crypto-Conditions <https://tools.ietf.org/html/draft-thomas-crypto-conditions-02>`_).
 
-Inputs
-------
+The simplest supported condition is a simple signature condition.
+Such a condition could be stated as,
+"You can satisfy this condition
+if you send me a message and a cryptographic signature of that message,
+produced using the private key corresponding to this public key."
+The public key is put in the output.
+BigchainDB currently only supports ED25519 signatures.
 
-One Current Owner
-`````````````````
-
-If there is only one *current owner*, the fulfillment will be a simple signature fulfillment (i.e. containing just one signature).
-
-.. code-block:: json
+A more complex condition can be composed by using n simple signature conditions as inputs to an m-of-n threshold condition (a logic gate which outputs TRUE if and only if m or more inputs are TRUE). If there are n inputs to a threshold condition:
 
-    {
-        "owners_before": ["<public key of the owner before the transaction happened>"],
-        "fulfillment": "cf:4:RxFzIE679tFBk8zwEgizhmTuciAylvTUwy6EL6ehddHFJOhK5F4IjwQ1xLu2oQK9iyRCZJdfWAefZVjTt3DeG5j2exqxpGliOPYseNkRAWEakqJ_UrCwgnj92dnFRAEE",
-        "fulfills": {
-            "output_index": 0,
-            "transaction_id": "11b3e7d893cc5fdfcf1a1706809c7def290a3b10b0bef6525d10b024649c42d3"
-        }
-    }
+* 1-of-n is the same as a logical OR of all the inputs
+* n-of-n is the same as a logical AND of all the inputs
 
+For example, one could create a condition requiring m (of n) signatures before their asset can be transferred.
 
-See the reference on :ref:`inputs <Input>` for descriptions of the meaning of each field.
+The (single) output of a threshold condition can be used as one of the inputs of other threshold conditions. This means that one can combine threshold conditions to build complex logical expressions, e.g. (x OR y) AND (u OR v).
 
-Multiple Current Owners
-```````````````````````
+When one creates a condition, one can calculate its fulfillment length (e.g.
+96). The more complex the condition, the larger its fulfillment length will be.
+A BigchainDB federation can put an upper limit on the complexity of the
+conditions, either directly by setting a maximum allowed fulfillment length,
+or
+`indirectly <https://github.com/bigchaindb/bigchaindb/issues/356#issuecomment-288085251>`_
+by :ref:`setting a maximum allowed transaction size <Enforcing a Max Transaction Size>`
+which would limit
+the overall complexity accross all inputs and outputs of a transaction.
+Note: At the time of writing, there was no configuration setting
+to set a maximum allowed fulfillment length,
+so the only real option was to
+:ref:`set a maximum allowed transaction size <Enforcing a Max Transaction Size>`.
```

Comment:
Oh wait, this PR is merged already. I'll make a small new PR to make that change.